README.md
"# Automatic-Attendance-System" 
"# Final_Project" 


changes.md
#changes being made.

1. Changed Image path 
2. commented line no.12 in main.py
3. changed images path in student.py
"student.py:19: DeprecationWarning: ANTIALIAS is deprecated and will be removed in Pillow 10 (2023-07-01). Use Resampling.LANCZOS instead."
4. Fixed it
5. added pass in face_recognition.py line no.59


help.py
from tkinter import*
from tkinter import ttk
from PIL import Image, ImageTk
from tkinter import messagebox
import mysql.connector
import cv2

class Help :
    def __init__(self,root):
        self.root=root
        self.root. geometry ( "1530x790+0+0")
        self.root.title("face Recogniton System")

        title_lbl = Label(self.root, text="HELP DESK ",font=("helvetica", 25,"bold"),bg="white",fg="green")
        title_lbl.place(x=0,y=0,width=1530,height=45)
# img 1
        img_top=Image.open(r"Images\help2.jpg")
        img_top=img_top.resize((1530,710),Image.Resampling.LANCZOS) 
        self.photoimg_top=ImageTk.PhotoImage(img_top)

        f_lb1=Label(self.root,image=self.photoimg_top)
        f_lb1.place(x=0,y=40,width=1530,height=720)

        help_lbl = Label(f_lb1, text="Email: paritoshpal7796@gmail.com",font=("helvetica", 20,"bold"),bg="beige",fg="black")
        help_lbl.place(x=550,y=460)

        help_lb2 = Label(f_lb1, text="Contact: 9403191396",font=("helvetica", 20,"bold"),bg="beige",fg="black")
        help_lb2.place(x=550,y=499)




        
if __name__ == "__main__":
    root=Tk()
    obj=Help(root)
    root.mainloop()



main.py
from cProfile import label
from cgitb import handler, text
from multiprocessing import parent_process
from time import strftime
import tkinter
from tkinter import*
from tkinter import ttk
from tkinter import font
import os
from turtle import back, title 
from PIL import Image,ImageTk
from datetime import datetime
from student import Student
from attendance import Attendance
from train import Train
#commenting it coz it has indentation issues
from face_recognition import Face_Recognition 
from help import Help



class Face_Recognition_System:
    def __init__(self,root):
        self.root=root
        self.root.geometry("1530x790+0+0")
        self.root.title("Face Recognition System")

        # Img1
        img = Image.open(r"Images\banner.jpg")
        img = img.resize((1530,130))
        self.photoimg = ImageTk.PhotoImage(img)


        f_lbl = Label(self.root,image=self.photoimg)
        f_lbl.place(x=0,y=0,width=1530,height=130)


        # # Img 2
        # img1 = Image.open(r"Images\FaceRecog.jpg")
        # img1 = img1.resize((500,130))
        # self.photoimg1 = ImageTk.PhotoImage(img1)


        # f_lbl = Label(self.root,image=self.photoimg1)
        # f_lbl.place(x=500,y=0,width=500,height=130)


        # # img 3
        # img2 = Image.open(r"Images\database.png")
        # img2 = img2.resize((500,130))
        # self.photoimg2 = ImageTk.PhotoImage(img2)


        # f_lbl = Label(self.root,image=self.photoimg2)
        # f_lbl.place(x=1000,y=0,width=550,height=130)

        # img bg
        img3 = Image.open(r"Images\help2.jpg")
        img3 = img3.resize((1530,710))
        self.photoimg3 = ImageTk.PhotoImage(img3)


        bg_img = Label(self.root,image=self.photoimg3)
        bg_img.place(x=0,y=130,width=1530,height=710)

        title_lbl = Label(bg_img, text="Face Recognition Attendance System",font=("helvetica", 25,"bold"),bg="Beige",fg="black")
        title_lbl.place(x=0,y=0,width=1530,height=45)

        def  time(): 
            string = strftime('%H:%M:%S %p')
            lb1.config(text=string)
            lb1.after(1000, time)

        lb1 = Label(title_lbl, font=("time new roman",14,"bold"),background='white',foreground='blue')
        lb1.place(x=0,y=0,width=110,height=50)
        time()


        # Student Button
        img4 = Image.open(r"Images\student.jpg")
        img4 = img4.resize((220,220))
        self.photoimg4= ImageTk.PhotoImage(img4)

        b1=Button(bg_img,image=self.photoimg4,command=self.student_details,cursor="hand2")
        b1.place(x=200,y=100,width=220,height=220)

        b1_1=Button(bg_img,text="Student Details",command=self.student_details,cursor="hand2",font=("helvetica", 15,"bold"),bg="Beige",fg="black")
        b1_1.place(x=200,y=300,width=220,height=40)

        # Detect Face
        img5 = Image.open(r"Images\detectFace.png")
        img5 = img5.resize((220,220))
        self.photoimg5= ImageTk.PhotoImage(img5)

        b1=Button(bg_img,image=self.photoimg5,cursor="hand2",command=self.face_data)
        b1.place(x=500,y=100,width=220,height=220)

        b1_1=Button(bg_img,text="Face Recognition",cursor="hand2",command=self.face_data,font=("helvetica", 15,"bold"),bg="Beige",fg="black")
        b1_1.place(x=500,y=300,width=220,height=40)


        # Take Attendance !
        img6 = Image.open(r"Images\Attendance.png")
        img6 = img6.resize((220,220))
        self.photoimg6= ImageTk.PhotoImage(img6)

        b1=Button(bg_img,image=self.photoimg6,cursor="hand2",command=self.atten_data)
        b1.place(x=800,y=100,width=220,height=220)

        b1_1=Button(bg_img,text="Attendance",cursor="hand2",command=self.atten_data,font=("helvetica", 15,"bold"),bg="Beige",fg="black")
        b1_1.place(x=800,y=300,width=220,height=40)


        # Help Attendance !
        img7 = Image.open(r"Images\help.jpg")
        img7 = img7.resize((220,220))
        self.photoimg7= ImageTk.PhotoImage(img7)

        b1=Button(bg_img,image=self.photoimg7,cursor="hand2",command=self.help_data)
        b1.place(x=1100,y=100,width=220,height=220)

        b1_1=Button(bg_img,text="Help Desk ",cursor="hand2",command=self.help_data,font=("helvetica", 15,"bold"),bg="Beige",fg="black")
        b1_1.place(x=1100,y=300,width=220,height=40)


        # Train the Image
        img8 = Image.open(r"Images\Tr.jpg")
        img8 = img8.resize((220,220))
        self.photoimg8= ImageTk.PhotoImage(img8)

        b1=Button(bg_img,image=self.photoimg8,cursor="hand2",command=self.train_data)
        b1.place(x=200,y=380,width=220,height=220)

        b1_1=Button(bg_img,text="Train Data ",cursor="hand2",command=self.train_data,font=("helvetica", 15,"bold"),bg="Beige",fg="black")
        b1_1.place(x=200,y=570,width=220,height=40)


        # Images Data Set...
        img9 = Image.open(r"Images\Dataset.jpg")
        img9 = img9.resize((220,220),Image.ANTIALIAS)
        self.photoimg9= ImageTk.PhotoImage(img9)

        b1=Button(bg_img,image=self.photoimg9,cursor="hand2",command=self.open_img)
        b1.place(x=500,y=380,width=220,height=220)

        b1_1=Button(bg_img,text="Photos",cursor="hand2",command=self.open_img,font=("helvetica", 15,"bold"),bg="Beige",fg="black")
        b1_1.place(x=500,y=570,width=220,height=40)


 # Exit
        img10 = Image.open(r"Images\Exit.jpg")
        img10 = img10.resize((220,220))
        self.photoimg10= ImageTk.PhotoImage(img10)

        b1=Button(bg_img,image=self.photoimg10,cursor="hand2")
        b1.place(x=800,y=380,width=220,height=220)

        b1_1=Button(bg_img,text="Exit ",cursor="hand2",font=("helvetica", 15,"bold"),bg="Beige",fg="black")
        b1_1.place(x=800,y=570,width=220,height=40)

    def open_img(self):
        os.startfile("Data")

    # def iExit(self):
    #     self.iExit=tkinter.askyesno("Face Recognition","Are you sure to exit ",parent=self.root)
    #     if self.iExit > 0:
    #         self.root.destroy()
    #     else:
    #         return 
    # IIIIIIIIMMMMMMMMMPPPPPPPPP!!!!!!!!!!!!!!!!!!!!!!
    #  commenting beacause error .... tkinter has no attribute askyesno
            
            
        
    #============= Function buttons===== 
    def student_details(self):
        self.new_window = Toplevel(self.root)
        self.app = Student(self.new_window)

    def train_data(self):
        self.new_window=Toplevel(self.root)
        self.app=Train(self.new_window)

    def face_data(self):
        self.new_window=Toplevel(self.root)
        self.app=Face_Recognition(self.new_window)

    def help_data(self):
        self.new_window=Toplevel(self.root)
        self.app=Help(self.new_window)

    def atten_data(self):
        self.new_window=Toplevel(self.root)
        self.app=Attendance(self.new_window)


       
                #  self.app=Train(self.new_window)
               














if __name__ == "__main__":
    root=Tk()
    obj=Face_Recognition_System(root)
    root.mainloop()

train.py
from tkinter import*
from tkinter import ttk
from PIL import Image, ImageTk
from tkinter import messagebox
import mysql.connector
import cv2
import os
import numpy as np
# 37


class Train:
    def __init__(self,root):
        self.root=root
        self.root. geometry ( "1530x790+0+0")
        self.root.title("face Recogniton System")

 # Img1
        img = Image.open(r"Images\banner.jpg")
        img = img.resize((1530,130))
        self.photoimg = ImageTk.PhotoImage(img)


        f_lbl = Label(self.root,image=self.photoimg)
        f_lbl.place(x=0,y=0,width=1530,height=130)

        img3 = Image.open(r"Images\help2.jpg")
        img3 = img3.resize((1530,710))
        self.photoimg3 = ImageTk.PhotoImage(img3)


        bg_img = Label(self.root,image=self.photoimg3)
        bg_img.place(x=0,y=130,width=1530,height=710)

        title_lbl = Label(bg_img, text="TRAIN DATA SET",font=("helvetica", 25,"bold"),bg="Beige",fg="black")
        title_lbl.place(x=0,y=0,width=1530,height=45)

        # title_lbl = Label(self.root, text="TRAIN DATA SET ",font=("helvetica", 25,"bold"),bg="white",fg="maroon")
        # title_lbl.place(x=0,y=0,width=1530,height=45)

        # img_top=Image.open(r"Images\banner.jpg")
        # img_top=img_top.resize((1530,325),Image.ANTIALIAS) 
        # self.photoimg_top=ImageTk.PhotoImage(img_top)

        # f_lb1=Label(self.root,image=self.photoimg_top)
        # f_lb1.place(x=5,y=55,width=1530,height=325)
# Buttonn
        b1_1=Button(self.root,text="TRAIN DATA",command=self.train_classifier ,cursor="hand2",font=("helvetica", 30,"bold"),bg="beige",fg="crimson")
        b1_1.place(x=0,y=380,width=550,height=60)

        # img_bottom=Image.open(r"Images\help2.jpg")
        # img_bottom=img_bottom.resize((1530,325),Image.ANTIALIAS) 
        # self.photoimg_bottom=ImageTk.PhotoImage(img_bottom)

        # f_lb1=Label(self.root,image=self.photoimg_bottom)
        # f_lb1.place(x=5,y=440,width=1530,height=325)


    def train_classifier(self):
        data_dir = ("data")
        path=[os.path.join(data_dir,file) for file in os.listdir(data_dir)]


        faces=[]
        ids=[]

        for image in path:
            img = Image.open(image).convert('L') #Converting image to grayscale
            imageNp=np.array(img,'uint8')
            id=int(os.path.split(image)[1].split('.')[1])

            faces.append(imageNp)
            ids.append(id)
            cv2.imshow("Training",imageNp)
            cv2.waitKey(1)==13
        ids=np.array(ids)

    # ////Train the classifier and save 

        clf=cv2.face.LBPHFaceRecognizer_create()
        clf.train(faces,ids)
        clf.write("classifier.xml")
        cv2.destroyAllWindows()
        messagebox.showinfo("Result"," Training data set Completed!!")






if __name__ == "__main__":
    root=Tk()
    obj=Train(root)
    root.mainloop()





face_recognition.py
#import codeop
#from email.mime import image
#import itertools
#import predicate
#from pyexpat import features
from tkinter import*
from tkinter import ttk
from PIL import Image, ImageTk
from tkinter import messagebox
import mysql.connector
import cv2
import os
import numpy as np
from time import strftime
from datetime import datetime



class Face_Recognition:
    def __init__(self,root):
        self.root=root
        self.root. geometry ( "1530x790+0+0")
        self.root.title("face Recogniton System")

        title_lbl = Label(self.root, text="FACE RECOGNITION ",font=("helvetica", 25,"bold"),bg="white",fg="green")
        title_lbl.place(x=0,y=0,width=1530,height=45)
# # Img1
        img = Image.open(r"Images\banner.jpg")
        img = img.resize((1530,130))
        self.photoimg = ImageTk.PhotoImage(img)


        f_lbl = Label(self.root,image=self.photoimg)
        f_lbl.place(x=0,y=0,width=1530,height=130)


        img3 = Image.open(r"Images\help2.jpg")
        img3 = img3.resize((1530,710))
        self.photoimg3 = ImageTk.PhotoImage(img3)


        bg_img = Label(self.root,image=self.photoimg3)
        bg_img.place(x=0,y=130,width=1530,height=710)

        title_lbl = Label(bg_img, text="FACE RECOGNITION ",font=("helvetica", 25,"bold"),bg="Beige",fg="black")
        title_lbl.place(x=0,y=0,width=1530,height=45)

# img 2
        img_bottom=Image.open(r"Images\face_recogR.jpg")
        img_bottom=img_bottom.resize((950,325),Image.Resampling.LANCZOS) 
        self.photoimg_bottom=ImageTk.PhotoImage(img_bottom)

        f_lb1=Label(self.root,image=self.photoimg_bottom)
        f_lb1.place(x=425,y=185,width=950,height=550)

        b1_1=Button(f_lb1,text="START FACE RECOGNITION",command=self.face_recog,cursor="hand2",font=("helvetica", 18,"bold"),bg="black",fg="white")
        b1_1.place(x=50,y=445,width=400,height=40)



    #================ Attendacne ========

    def mark_attendance(self,i,r,n,d):
        with open("bvcoel.csv","r+",newline='\n') as f:
            myDataList=f.readlines()
            name_list=[]
            for line in myDataList:
                entry=line.split((","))
                name_list.append(entry[0])
            #this is so that it doesnt captures multiple times one person
            if((i not in name_list) and (r not in name_list) and (n not in name_list) and (d not in name_list)):
                now=datetime.now()
                d1=now.strftime('%d/%m/%Y')
                dtString=now.strftime('%H:%M:%S')
                f.writelines(f"\n{i},{r},{n},{d},{dtString},{d1},Present")



    



    #========= face recognition funciton =====

    def face_recog(self):
        def draw_boundray(img,classifier,scaleFactor,minNeighbors,color,text,clf):
            gray_image=cv2.cvtColor(img,cv2.COLOR_BGR2GRAY)
            features=classifier.detectMultiScale(gray_image,scaleFactor,minNeighbors)

            coord=[]

            for (x,y,w,h) in features:
                cv2.rectangle(img,(x,y),(x+w,y+h),(0,255,0),3)
                id,predict = clf.predict(gray_image[y:y+h,x:x+w])
                confidence=int((100*(1-predict/300)))

                conn=mysql.connector.connect(host="localhost",username="root",password="admin")
                my_cursor=conn.cursor()


                my_cursor.execute("select Name from face_recognizer.student where Student_id="+str(id))
                n=my_cursor.fetchone()     
                n.sort()         
                n="+".join(n)

                my_cursor.execute("select Roll from face_recognizer.student where Student_id="+str(id))
                r=my_cursor.fetchone()
                r="+".join(r)

                my_cursor.execute("select Dep from face_recognizer.student where Student_id="+str(id))
                d=my_cursor.fetchone()
                d="+".join(d)

                my_cursor.execute("select Student_id from face_recognizer.student where Student_id="+str(id))
                i=my_cursor.fetchone()
                i="+".join(i)




                if confidence>77:
                    cv2.putText(img,f"ID: {i}",(x,y-75),cv2.FONT_HERSHEY_COMPLEX,0.8,(255,255,255),3)
                    cv2.putText(img,f"Roll: {r}",(x,y-55),cv2.FONT_HERSHEY_COMPLEX,0.8,(255,255,255),3)
                    cv2.putText(img,f"Name: {n}",(x,y-30),cv2.FONT_HERSHEY_COMPLEX,0.8,(255,255,255),3)
                    cv2.putText(img,f"Department: {d}",(x,y-5),cv2.FONT_HERSHEY_COMPLEX,0.8,(255,255,255),3)
                    self.mark_attendance(i,r,n,d)
                else:
                    cv2.rectangle(img,(x,y),(x+w,y+h),(0,0,255),3)
                    cv2.putText(img,"Unknown Face: {d}",(x,y-5),cv2.FONT_HERSHEY_COMPLEX,0.8,(255,255,255),3)

                coord=[x,y,w,y]

            return coord

        def recognize(img,clf,faceCascade):
            coord=draw_boundray(img,faceCascade,1.1,10,(255,25,255),'Face',clf)
            return img 

        #6th part 37.10

        faceCascade=cv2.CascadeClassifier("haarcascade_frontalface_default.xml")
        clf=cv2.face.LBPHFaceRecognizer_create()
        clf.read("classifier.xml")

        video_cap=cv2.VideoCapture(0)

        while True:
            ret,img=video_cap.read()
            img=recognize(img,clf,faceCascade)
            cv2.imshow("Welcome to Face Recognition",img)

            if cv2.waitKey(1)==13:
                break
        video_cap.release()
        cv2.destroyAllWindows()







        

if __name__ == "__main__":
    root=Tk()
    obj=Face_Recognition(root)
    root.mainloop()



attendance.py
from cProfile import label
from cgitb import handler, text
from lzma import MF_BT2
from tkinter import*
from tkinter import ttk
from tkinter import font
from turtle import title, width 
from PIL import Image,ImageTk
# from student import Student
from tkinter import messagebox
import mysql.connector
import cv2 


class Attendance:
    def __init__(self,root):
        self.root=root
        self.root.geometry("1530x700+0+0")
        self.root.title("Face Recognition System")

#          # Img1
#         img = Image.open(r"Images\\attendance.png")
#         img = img.resize((800,200),Image.Resampling.LANCZOS)
#         self.photoimg = ImageTk.PhotoImage(img)


#         f_lbl = Label(self.root,image=self.photoimg)
#         f_lbl.place(x=0,y=0,width=800,height=200)


# # Img 2
#         img1 = Image.open(r"Images\\attendance.png")
#         img1 = img1.resize((730,200),Image.Resampling.LANCZOS)
#         self.photoimg1 = ImageTk.PhotoImage(img1)


#         f_lbl = Label(self.root,image=self.photoimg1)
#         f_lbl.place(x=800,y=0,width=730,height=200)

 # Img1
        img = Image.open(r"Images\banner.jpg")
        img = img.resize((1530,130))
        self.photoimg = ImageTk.PhotoImage(img)


        f_lbl = Label(self.root,image=self.photoimg)
        f_lbl.place(x=0,y=0,width=1530,height=130)

        img3 = Image.open(r"Images\help2.jpg")
        img3 = img3.resize((1530,710))
        self.photoimg3 = ImageTk.PhotoImage(img3)


        bg_img = Label(self.root,image=self.photoimg3)
        bg_img.place(x=0,y=130,width=1530,height=710)

        title_lbl = Label(bg_img, text="ATTENDANCE MANAGEMENT SYSTEM",font=("helvetica", 25,"bold"),bg="white",fg="crimson")
        title_lbl.place(x=0,y=0,width=1530,height=45)


        main_frame=Frame(bg_img,bd=2,bg="white")
        main_frame.place(x=0,y=50,width=1520,height=600)

         # #left label frame
        Left_frame=LabelFrame(main_frame,bd=2,bg="white",relief=RIDGE,text="Student Attendance Details",font=("time new roman",12,"bold"))
        Left_frame.place(x=0,y=5,width=720,height=580)

        img_left=Image.open(r"Images\left.jpg")
        img_left=img_left.resize((720,130),Image.Resampling.LANCZOS) 
        self.photoimg_left=ImageTk.PhotoImage(img_left)

        f_lb1=Label(Left_frame,image=self.photoimg_left)
        f_lb1.place(x=5,y=0,width=700,height=130)

        left_inside_frame=Frame(Left_frame,bd=2,relief=RIDGE,bg="white")
        left_inside_frame.place(x=0,y=135,width=710,height=370)

        # Labels and Entries 

         #attendance id 
        attendanceId_label=Label(left_inside_frame,text="Attendance ID:",font=("time new roman",12,"bold"),bg="white")
        attendanceId_label.grid(row=0,column=0,padx=10,pady=5,sticky=W)

        studentID_entry=ttk.Entry(left_inside_frame,width=20,font=("times new roman",13,"bold"))
        studentID_entry.grid(row=0,column=1,padx=10,pady=5,sticky=W)

        # Name
        rollLabel=Label(left_inside_frame,text="Roll:",font=("time new roman",12,"bold"),bg="white")
        rollLabel.grid(row=0,column=2,padx=8)

        atten_roll=ttk.Entry(left_inside_frame,width=20,font=("times new roman",13,"bold"))
        atten_roll.grid(row=0,column=3,padx=10,pady=8)

        # Date
        nameLabel=Label(left_inside_frame,text="Name:",font=("time new roman",12,"bold"),bg="white")
        nameLabel.grid(row=1,column=0)

        atten_name=ttk.Entry(left_inside_frame,width=20,font=("times new roman",13,"bold"))
        atten_name.grid(row=1,column=1, pady=7)

        # DEpt 
        depLabel=Label(left_inside_frame,text="Department:",font=("time new roman",12,"bold"),bg="white")
        depLabel.grid(row=1,column=2,padx=7)

        atten_dep=ttk.Entry(left_inside_frame,width=20,font=("times new roman",13,"bold"))
        atten_dep.grid(row=1,column=3,padx=7)

        #Time 
        timeLabel=Label(left_inside_frame,text="Time:",font=("time new roman",12,"bold"),bg="white")
        timeLabel.grid(row=2,column=0,padx=7)

        atten_time=ttk.Entry(left_inside_frame,width=20,font=("times new roman",13,"bold"))
        atten_time.grid(row=2,column=1,padx=7)

        # Date
        dateLabel=Label(left_inside_frame,text="Date:",font=("time new roman",12,"bold"),bg="white")
        dateLabel.grid(row=2,column=2)

        atten_date=ttk.Entry(left_inside_frame,width=20,font=("times new roman",13,"bold"))
        atten_date.grid(row=2,column=3,pady=7)
        
        # Attendance
        attendanceLabel=Label(left_inside_frame,text="Attendance Status:",font=("time new roman",12,"bold"),bg="white")
        attendanceLabel.grid(row=3,column=0)

        self.atten_status=ttk.Combobox(left_inside_frame, width=20,font="comicsansns 11 bold", state="readonly" )
        self.atten_status["values"]=("Status", "Present", "Absent")
        self.atten_status.grid(row=3, column=1,pady=7)
        self.atten_status.current(0)

         #bbuttons frame
        btn_frame=Frame(left_inside_frame,bd=2,relief=RIDGE,bg="white")
        btn_frame.place(x=0,y=300,width=715,height=70)

        save_btn=Button(btn_frame,text="Import csv",width=17,font=("times new roman",13,"bold"),bg="Beige",fg="black")
        save_btn.grid(row=0,column=0)

        update_btn=Button(btn_frame,text="Export csv",width=17,font=("times new roman",13,"bold"),bg="Beige",fg="black")
        update_btn.grid(row=0,column=1)

        delete_btn=Button(btn_frame,text="Update",width=17,font=("times new roman",13,"bold"),bg="Beige",fg="black")
        delete_btn.grid(row=0,column=2)

        reset_btn=Button(btn_frame,text="Reset",width=17,font=("times new roman",13,"bold"),bg="Beige",fg="black")
        reset_btn.grid(row=0,column=3)

          # #right frame
        Right_frame=LabelFrame(main_frame,bd=2,bg="white",relief=RIDGE,text="Attendance Details",font=("time new roman",12,"bold"))
        Right_frame.place(x=750,y=10,width=735,height=580)

        # img_right=Image.open(r"Images\right.png")
        # img_right=img_right.resize((700,130),Image.Resampling.LANCZOS) 
        # self.photoimg_right=ImageTk.PhotoImage(img_right)

        # f_lb1=Label(Right_frame,image=self.photoimg_right)
        # f_lb1.place(x=5,y=5,width=700,height=130)

        #   Right inside frame  table frame 

        table_frame=Frame(Right_frame,bd=2,relief=RIDGE,bg="white")
        table_frame.place(x=5,y=5,width=710,height=445)

        # Scroll bar and TABLE #######################
        scroll_x = ttk.Scrollbar(table_frame, orient=HORIZONTAL)
        scroll_y = ttk.Scrollbar(table_frame, orient=VERTICAL)

        self.AttendanceReportTable = ttk.Treeview(table_frame, column=("id","roll","name","department","time","date","attendance"),xscrollcommand=scroll_x.set, yscrollcommand=scroll_y.set)

        scroll_x.pack(side=BOTTOM,fill=X)
        scroll_y.pack(side=RIGHT,fill=Y)

        scroll_x.config(command=self.AttendanceReportTable.xview)
        scroll_y.config(command=self.AttendanceReportTable.yview)

        self.AttendanceReportTable.heading("id",text="Attendance ID")
        self.AttendanceReportTable.heading("roll",text="Roll")
        self.AttendanceReportTable.heading("name",text="Name")
        self.AttendanceReportTable.heading("department",text="Department")
        self.AttendanceReportTable.heading("time",text="Time")
        self.AttendanceReportTable.heading("date",text="Date")
        self.AttendanceReportTable.heading("attendance",text="Attendance")

        self.AttendanceReportTable["show"]="headings"

    
        self.AttendanceReportTable.column("id",width=100)
        self.AttendanceReportTable.column("roll", width=100)
        self.AttendanceReportTable.column("name",width=100)
        self.AttendanceReportTable.column("department",width=100)
        self.AttendanceReportTable.column("time",width=100)
        self.AttendanceReportTable.column("date",width=100)
        self.AttendanceReportTable.column("attendance",width=100)
    

        self.AttendanceReportTable.pack(fill=BOTH, expand=1)

        

       




if __name__ == "__main__":
    root=Tk()
    obj=Attendance(root)
    root.mainloop()


